extensions/amp-carousel/0.1/scrollable-carousel.js

```diff
@@ -126,6 +139,57 @@ export class AmpScrollableCarousel extends BaseCarousel {
     }
   }
 
+  /**
+   * Scrolls to the slide at the given slide index.
+   * @param {number} index
+   * @private
+   */
+  goToSlide_(index) {
+    const noOfSlides = this.cells_.length;
+
+    if (!isFinite(index) || index < 0 || index >= noOfSlides) {
+      this.user().error(TAG, 'Invalid [slide] value: %s', index);
+      return Promise.resolve();
+    }
+
+    const oldPos = this.pos_;
+
+    const measureSlidePosition = state => {
+      state.newPos = this.getPosForSlideIndex_(index);
+    };
+
+    const mutateSlidePosition = state => {
+      if (state.newPos == oldPos) {
+        return;
+      }
+      /** @const {!TransitionDef<number>} */
+      const interpolate = numeric(oldPos, state.newPos);
+      const duration = 200;
+      const curve = 'ease-in-out';
+      Animation.animate(this.element, pos => {
+        this.container_./*OK*/scrollLeft = interpolate(pos);
+      }, duration, curve).thenAlways(() => {
+        this.commitSwitch_(state.newPos);
+      });
+    };
+
+    this.vsync_.run({
```

Comment:
`VSync` will be a deprecated API soon (I'm actively removing every use). Please use `BaseElement#measureMutateElement`, and note that `state` is not supported (it doesn't help you in the this situation anyways).

Reply:
Added to tomorrow's deep dive.